### ISSUE TITLE
"Unknown Error" while decompiling RISC-V binary

### ISSUE THREAD

**Issue by bruce30262** [NationalSecurityAgency/ghidra#2466]
## Describe the bug
I've been testing the RISC-V decompiler with the following RISC-V binaries: 

![rrr](https://user-images.githubusercontent.com/4478350/99139314-89811f00-2672-11eb-90f2-632c1038ed36.jpg)

![hitree](https://user-images.githubusercontent.com/4478350/99139322-969e0e00-2672-11eb-9308-b4f3b8978dfb.jpg)

Both of them are ELF 64-bit, UCB RISC-V, statically linked binary
( To download : Right click --> Save link as --> Remove the `.jpg` extension )

Ghidra failed to decompile some of the functions ( e.g. `main` ) in those binaries, showing "Unknown Error" in the decompile window.  

For example, in the `rrr` binary, ghidra failed to decompile the following functions :  
* `__do_global_dtors_aux`  
* `print_flag`  
* `game`  
* `main`  

The other functions seems to be decompiled normally ( did not check all of them though ).  

Same errors occur in `hitree` as well. At least the `main` function is not being decompiled.

## To Reproduce
1.  Create a new non-shared project, drag those binaries into the project.
2.  Ghidra will detect the language as `RISCV:LE:64:RV64I (1.2)` ( on both binaries ).
3.  Click OK and start analyze ( all options are default ).
4.  Goto the `main` funtion and check the decompile window.
5.  Decompile window will show "Unknown Error".

## Expected behavior
Expected the decompiler work flawlessly. 

## Screenshots

rrr's main function: 
<img width="950" alt="rrr_main" src="https://user-images.githubusercontent.com/4478350/99139699-18436b00-2676-11eb-9af5-7acb54ab30e8.PNG">

hitree's main function:  
<img width="899" alt="hitree_main" src="https://user-images.githubusercontent.com/4478350/99139718-327d4900-2676-11eb-8542-215a94029324.PNG">

## Attachments
[application.log](https://github.com/NationalSecurityAgency/ghidra/files/5540211/application.log)

## Environment
 - OS: Windows 10
 - Java Version: 11.0.2
 - Ghidra Version: 9.2 ( 2020/11/13 )

**Comment by mumbel** [comment]
Ah, looks like the issue is `gp` most likely.  The RISC-V java analyzer code definitely needs improvement, I only (barely) implemented code to do this.  Calculate the `gp` off the `auipc` and `addi` in the `entry` function and in functions where `gp` is used you can right-click and set register values (or ctrl-A and do the whole program).

**Comment by bruce30262** [comment]
@mumbel thanks for providing the solution !  

Here for the above two binaries, I created a [simple workaround script](https://gist.github.com/bruce30262/c075e7ea623b8997f98c8137b8216b46) after learning some ghidra python. The script will calculate the value of `gp` register base on the code in `entry()` and apply the value to all functions.  

However I think this only works in the `RV64I` language, since I found [another RISC-V binary](https://github.com/phra/metasploit2020-five-of-hearts/blob/master/binary) ( `RV64GC` language ) which will set the `gp` register in a non-entry function ( `entry()` will call that function though ).  

I think a better way to fix this is to emulate the `entry()` function with [EmulatorHelper](https://ghidra.re/ghidra_docs/api/ghidra/app/emulator/EmulatorHelper.html) ( till it jumps to `main()` ). Once the value of `gp` register is confirmed, it then can be applied to the entire binary.
